src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,191 @@
-from typing import Tuple, Union
+from typing import Optional, Tuple, Union
+
+import numpy as np
 
 __all__ = ["TimeSteppingControl"]
 
 
 class TimeSteppingControl:
-    """Parent class for iteration-based time stepping control routine."""
+    """Parent class for iteration-based time stepping control routine.
+
+    Parameters:
+        schedule: List containing the target times for the simulation.
+            The time-stepping algorithm will adapt the time step so that the target
+            times are guaranteed to be hit/reached. The list must contain minimally
+            two elements, corresponding to the initial and final simulation times.
+            Lists of length > 2 must contain strictly increasing times.
+            Examples of VALID inputs are:
+                [0, 1]
+                [0, 10, 30, 50]
+                [0, 1*pp.HOUR, 3*pp.HOUR].
+            Examples of INVALID inputs are:
+                [1]
+                [1, 0]
+                [0, 1, 1, 2]
+        dt_init: Initial time step. If `constant_dt` is True, then `dt_init` is taken as
+            the constant time step.
+        constant_dt: Whether to treat the time step as constant or not. If True, then
+            the time-stepping control algorithm is effectively bypassed. To be
+            precise, the algorithm won't adapt the time step in any situation.
```

Comment:
To be sure: If I set a constant time step of 2, and a schedule of [0, 5, 10], the 5 will be ignored? I think this fair, I just want to understand.